src/node.cpp

```diff
@@ -529,7 +533,9 @@ bool Node::serialize(string* d)
     char hasLinkCreationTs = plink ? 1 : 0;
     d->append((char*)&hasLinkCreationTs, 1);
 
-    d->append("\0\0\0\0\0", 6);
+    d->append((char*)&mSyncable, sizeof(mSyncable));
```

Comment:
Can we switch Node's serialize and unserialize to use CacheableReader/CacheableWriter which is much easier to maintain, since we are making changes in these functions?  thanks

Reply:
Wouldn't that require a lot of changes? I am not sure this is necessary right now

Reply:
I think it would make it a lot easier to review (as the current format is quite fiddly which makes it easy to miss problems), and easier to maintain in future.  It would also mean we automatically avoid issues with the sizes of types etc.